### PR TITLE
parse different formats of docstring / plain text references

### PR DESCRIPTION
**Pull request by de-code**
part of #15

this is adding support for the following formats:

sphinx:

```
:cite:`1987:nelson`
```

latex:

```
\cite{1987:nelson}
```

doxygen:

```
\cite 1987:nelson
```

For Sphinx and LaTex the references may be separated by a comma.

LaTex and Doxygen may have the issue of getting rejected by a linter (due to `\c`).

It is still not attempting to extract the purpose of the citation.

### CODE REVIEW

**Comment by de-code** [comment]
Example:

```console
$ python -m r2t2 --static --docstring examples/docstring_reference.py 
Referenced in: my_great_function
Source file: examples/docstring_reference.py
Line: 1
        [1] automatically parsed from docstring - https://doi.org/10.5281/zenodo.1185316

Referenced in: my_great_function_2
Source file: examples/docstring_reference.py
Line: 11
        [1] automatically parsed from docstring - 1987:nelson
```

**Comment by tinosulzer** [comment]
Not related to this issue directly, but how come `--docstring` needs to be passed as an argument? Shouldn't the default settings also parse docstrings?

**Comment by de-code** [comment]
> Not related to this issue directly, but how come `--docstring` needs to be passed as an argument? Shouldn't the default settings also parse docstrings?

Yes, I am not sure. I added it as a precaution. But it could certainly be made the default. I thought maybe it's good to get it tested more on real examples (in case it's causing more false positives).

File: examples/docstring_reference.py [hunk 1]
```diff
@@ -6,3 +6,10 @@ def my_great_function():
     http://doi.org/10.5281/zenodo.1185316
     """
     pass
+
+
+def my_great_function_2():
```
Comment (rgaiacs) [inline examples/docstring_reference.py:11]: Can we have different files for different features?

Comment (de-code) [reply examples/docstring_reference.py:11]: Sure, how about `docstring_doi_reference.py` and `docstring_sphinx_reference.py`?

**Pull request merged**